docs/core-concepts/run.mdx

```diff
@@ -7,71 +7,167 @@ toc_min_heading_level: 2
+
+<div style={{textAlign: 'center'}}>
+<img src={ImgSimpleETUDF} alt="Simple UDF fused.run() returning a geodataframe" style={{width: 400}} />
+</div>
+
+There are a few other ways to run a UDF:
+- [By name from your account](/core-concepts/run/#name-from-your-account)
+- [By public UDF name](/core-concepts/run/#public-udf-name)
+- [Using a token](/core-concepts/run/#token)
+- [Using a `udf` object](/core-concepts/run/#udf-object)
+- [From Github URL](/core-concepts/run/#github-url)
+- [From git commit hash (most recommended for teams)](/core-concepts/run/#git-commit-hash-recommended-for-most-stable-use-cases)
+
+### _Name_ (from your account)
+
+_When to use: When calling a UDF you made, from your own account._
+
+You can call any UDFs you have made simply by calling it by name (given when you [save a UDF](/core-concepts/write/#save-udfs)).
+
+(Note: This requires authentication)
+
+import ImgHelloWorldUDF from '@site/static/img/core-concepts/run-udfs/hello_world_bbox_udf.png';
 
-#### _Name_
+<div style={{textAlign: 'center'}}>
+<img src={ImgHelloWorldUDF} alt="Hello World UDF" style={{width: 700}} />
+</div>
 
-**UDF name.** Use this to run UDFs in your account. This requires authentication.
+This UDF can then be run in a notebook locally (granted that you have authenticated):
 
 ```python showLineNumbers
-fused.run("My_Udf")
+fused.run("Hello_World_bbox")
 ```
 
-#### _Tokens_
+import ImgRunHWUDF from '@site/static/img/core-concepts/run-udfs/running_hello_world_bbox_udf.png';
 
-**Public token.** UDFs in the [public repo](https://github.com/fusedio/udfs/tree/main) may be run for free and without authentication. Reference them by prefixing their name with `UDF_`. For example, the UDF `Get_Isochrone` is run with `UDF_Get_Isochrone`.
+<div style={{textAlign: 'center'}}>
+<img src={ImgRunHWUDF} alt="RunningHello World UDF" style={{width: 700}} />
+</div>
+
+### _Public UDF Name_
+
+_When to run: Whenever you want to run a public UDF for free from anywhere_
+
+Any UDF saved in the [public UDF repo](https://github.com/fusedio/udfs/tree/main) can be run for free.
+
+Reference them by prefixing their name with `UDF_`. For example, [the public UDF `Get_Isochrone`](https://github.com/fusedio/udfs/tree/main/public/Get_Isochrone) is run with `UDF_Get_Isochrone`:
 
 ```python showLineNumbers
 fused.run('UDF_Get_Isochrone')
 ```
 
+### _Token_
+
+_When to use: Whenever you want someone to be able to execute a UDF but might not want to share the code with them._
+
+You can get the token from a UDF either in [Workbench](/workbench/) (Save your UDF then go to "Settings" -> "Share" -> "Fused App") or returning the token in Python.
+
+Let's use the [same public Eiffel Tower UDF](https://github.com/fusedio/udfs/tree/main/public/Single_point_Eiffel_Tower) as an example:
+
+```python showLineNumbers
+from fused.api import FusedAPI
+api = FusedAPI()
+token = api.create_udf_access_token("Single_point_Eiffel_Tower").token
```

Comment:
Is this example expected to work when a reader of the docs (with access to fused) runs this themselves? (or just illustrative and you should replace it with a name of another UDF you own) 

I get "UDF not found" (because it looks for UDFs in my own account?)

Reply:
Interesting, I thought this worked for any public UDF 

I guess it makes sense that you can't make a token for a public UDF. 
I'll change this example by creating a new UDF inline in the example

Thank you for raising this up, good catch

Reply:
Addressed this here: https://github.com/fusedio/fused-docs/pull/290/commits/3e3bf3ca4ba1edc281afe98b923575f734bbed53